subworkflows/ebi-metagenomics/reads_bwamem2_decontamination/main.nf

```diff
@@ -0,0 +1,27 @@
+include { BWAMEM2_MEM     } from '../../../modules/ebi-metagenomics/bwamem2/mem/main'
+include { SAMTOOLS_BAM2FQ } from '../../../modules/ebi-metagenomics/samtools/bam2fq/main'
+
+
+workflow READS_BWAMEM2_DECONTAMINATION   {
+
+    take:
+    ch_reads       // channel (mandatory): [ val(meta), [ path(reads) ] ]
+    ch_reference   // channel (mandatory): [ val(meta2), path(ref_index) ] | meta2 contains the name of the reference genome
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    BWAMEM2_MEM( ch_reads, ch_reference )
+
+    ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())
+
+    SAMTOOLS_BAM2FQ(BWAMEM2_MEM.out.bam.map { meta, bam, bai -> [ meta, bam ] }, ch_reads.map { meta, reads -> meta.single_end == false } )
```

Comment:
Not sure how to control this inside the code. The easier way would be to run the pipeline once per (sample) line in the csv file instead of mixing inputs in the ch_reads.

Reply:
@Ales-ibt https://github.com/EBI-Metagenomics/motus_pipeline/blob/main/modules/decontamination.nf#L48
but again, not ideal solution

Reply:
I think with can use [multiMap](https://www.nextflow.io/docs/latest/operator.html#multimap)
```
BWAMEM2_MEM.out.bam
    .multiMap { meta, bam, bai ->
        bam: [ meta, bam ],
        split: meta.single_end == false
    }
    .set { ch_in_bam2fq }

SAMTOOLS_BAM2FQ( ch_in_bam2fq.bam, ch_in_bam2fq.split )
```
but, I'm not 100% sure this will work everytime (we could test it).

The other option is to modify SAMTOOLS_BAM2FQ -> [here](https://github.com/EBI-Metagenomics/nf-modules/blob/feature/reads_bwamem2_decontamination/modules/ebi-metagenomics/samtools/bam2fq/main.nf#L25)
```
if (split || meta.single_end == false)
```

Reply:
@Ales-ibt and @mberacochea decided to modify SAMTOOLS_BAM2FQ, to add the split as a third element in the input tuple. This makes it much easier to use. And it doesn't introduce much overhead for those using it, just map { meta, bam -> meta, bam, false | true }